components/TARGET_PSA/TARGET_TFM/COMPONENT_SPE/autogen/tfm_partition_defs.inc

```diff
@@ -0,0 +1,37 @@
+/* Copyright (c) 2017-2019 ARM Limited

```

Comment:
General question: any reason for going with `.inc` instead of `.h`? Mbed OS generally sticks with `.h` I believe.

Reply:
+1 
The _only_ other file that uses a `.inc` suffix in Mbed OS is `features/FEATURE_BLE/targets/TARGET_CORDIO_LL/thirdparty/uecc/asm_arm.inc`, and even that's third party.

Reply:
These are the partition header files used by TF-M.
TF-M actually generated these files according to their secure partitions manifests.
But because we use our own partitions we need to generate those headers ourselves.
So in order not to change TF-M imported code we keep the same suffix.

Reply:
> So in order not to change TF-M imported code we keep the same suffix.

Good enough for me!

Reply:
Actually, I'm not even sure if our build system picks up `.inc` files. Looking here: https://github.com/ARMmbed/mbed-os/blob/master/tools/resources/__init__.py#L413-L432

Seems like it won't be brought in as an include directory, does that sound right @theotherjimmy ?

Reply:
Actually it does work, #9221 runs TFM code on musca a1 target, and these sources include `.inc` files, see under `components/TARGET_PSA/TARGET_TFM/secure-fw/core/`.

Reply:
@mikisch81 I _believe_ that's working because there are other `.h` files present in the directory, so it is being picked up as an include path. If you removed all of the other `.h` files and just had `.inc` files, I don't believe the build system would keep that directory as an include path.

Reply:
One of the generated files is an `.h` file, so you are probably right, but all these files are generated together and shouldn't move or change afterwards.
We can ask TFM why the inc extension instead of regular `.h`.

Reply:
@bridadan We really need to avoid changing imported TF-M code, the only exceptions should be blocking issues (like bugs or porting issues).

Reply:
If this is a TF-M imported thing then I understand. We should get an issue filed to add `.inc` to our list of header file types in our build system.

Reply:
PR up for this issue: https://github.com/ARMmbed/mbed-os/pull/9715